### PR TITLE
Clean up Columns.astype & cudf.dtype

### PR DESCRIPTION
**Pull request by mroeschke**
## Description

- Able to remove `pandas_dtypes_alias_to_cudf_alias` by using `cudf.dtype` in `Column.astype`
- Simplified some branches in `Column.astype`

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/dtypes.py [hunk 1]
```diff
@@ -42,38 +42,36 @@ def dtype(arbitrary):
...
-    ) and cudf.api.types._is_pandas_nullable_extension_dtype(pd_dtype):
-        raise NotImplementedError("not supported")
-    try:
-        return dtype(pd_dtype.numpy_dtype)
-    except AttributeError:
-        if isinstance(pd_dtype, pd.CategoricalDtype):
-            return cudf.CategoricalDtype.from_pandas(pd_dtype)
+    if cudf.api.types._is_pandas_nullable_extension_dtype(pd_dtype):
+        if cudf.get_option("mode.pandas_compatible"):
+            raise NotImplementedError("not supported")
```
Comment (shwina) [inline python/cudf/cudf/core/dtypes.py:60]:
```suggestion
            raise NotImplementedError("Nullable types not supported in pandas compatibility mode")
```

**Review by shwina — APPROVED** [review]
Makes sense. One suggestion to improve an older error message.

**Review by isVoid — APPROVED** [review]
lgtm, pretty neat.

**Pull request merged**